### ISSUE TITLE
Problem with osc_elb resource

### ISSUE THREAD

**Issue by fbentouati** [remijouannet/terraform-provider-osc#1]
Hello,

I want use osc_elb resource and I faced crash below : 

```
Error: Error refreshing state: 1 error(s) occurred:

* osc_elb.elb-web: 1 error(s) occurred:

* osc_elb.elb-web: osc_elb.elb-web: unexpected EOF


panic: runtime error: invalid memory address or nil pointer dereference
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f540aa]
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: goroutine 506 [running]:
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: github.com/remijouannet/terraform-provider-osc/osc.resourceAwsElbRead(0xc4202521c0, 0x2197ba0, 0xc4202401c0, 0x0, 0x38cd9e0)
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /go/src/github.com/remijouannet/terraform-provider-osc/osc/resource_osc_elb.go:391 +0x76a
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: github.com/remijouannet/terraform-provider-osc/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc4202bdb60, 0xc420135130, 0x2197ba0, 0xc4202401c0, 0xc42029fbf0, 0x1031f01, 0x800204c1800)
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /go/src/github.com/remijouannet/terraform-provider-osc/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:321 +0x199
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: github.com/remijouannet/terraform-provider-osc/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc42020c0e0, 0xc4201350e0, 0xc420135130, 0x41a46c8, 0x0, 0x756ea1f8880eed)
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /go/src/github.com/remijouannet/terraform-provider-osc/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:284 +0x9a
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: github.com/remijouannet/terraform-provider-osc/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc42022d1e0, 0xc420582110, 0xc420582270, 0x0, 0x0)
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /go/src/github.com/remijouannet/terraform-provider-osc/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: reflect.Value.call(0xc4202644e0, 0xc42000cc98, 0x13, 0x275f250, 0x4, 0xc420043f20, 0x3, 0x3, 0x0, 0x0, ...)
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /usr/local/go/src/reflect/value.go:434 +0x906
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: reflect.Value.Call(0xc4202644e0, 0xc42000cc98, 0x13, 0xc420128f20, 0x3, 0x3, 0xc420235740, 0xc420235800, 0xc4205c40a0)
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /usr/local/go/src/reflect/value.go:302 +0xa4
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: net/rpc.(*service).call(0xc420306f00, 0xc420121950, 0xc420015130, 0xc420186480, 0xc4205fd660, 0x219a9a0, 0xc420582110, 0x16, 0x219a9e0, 0xc420582270, ...)
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /usr/local/go/src/net/rpc/server.go:381 +0x142
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3: created by net/rpc.(*Server).ServeCodec
2017-11-15T17:02:32.781+0100 [DEBUG] plugin.terraform-provider-osc_v0.3:        /usr/local/go/src/net/rpc/server.go:475 +0x36b
2017/11/15 17:02:32 [ERROR] root: eval: *terraform.EvalRefresh, err: osc_elb.elb-web: unexpected EOF
2017/11/15 17:02:32 [ERROR] root: eval: *terraform.EvalSequence, err: osc_elb.elb-web: unexpected EOF
2017/11/15 17:02:32 [TRACE] [walkRefresh] Exiting eval tree: osc_elb.elb-web
2017/11/15 17:02:32 [TRACE] dag/walk: upstream errored, not walking "provider.osc (close)"
2017/11/15 17:02:32 [TRACE] dag/walk: upstream errored, not walking "root"
2017-11-15T17:02:32.784+0100 [DEBUG] plugin: plugin process exited: path=/Users/fredericbentouati/.terraform.d/plugins/terraform-provider-osc_v0.3
2017/11/15 17:02:32 [DEBUG] plugin: waiting for all plugin processes to complete...
2017-11-15T17:02:32.790+0100 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2017-11-15T17:02:32.791+0100 [DEBUG] plugin.terraform: remote-exec-provisioner (internal) 2017/11/15 17:02:32 [DEBUG] plugin: waiting for all plugin processes to complete...
2017-11-15T17:02:32.793+0100 [DEBUG] plugin: plugin process exited: path=/usr/local/Cellar/terraform/0.10.8/bin/terraform



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

My terraform code : 
```

resource "osc_elb" "elb-web" {
  name               = "${var.project}-prod"
  security_groups = ["${osc_security_group.lb-web-sg.id}"]
  subnets         = ["${osc_subnet.subnet-public.id}"]

  listener {
    instance_port      = 80
    instance_protocol  = "HTTP"
    lb_port            = 80
    lb_protocol        = "HTTP"
  }

  health_check {
    healthy_threshold   = 2
    unhealthy_threshold = 2
    timeout             = 3
    target              = "HTTP:80/"
    interval            = 30
  }

  instances = ["${osc_instance.adm.id}"]

  tags {
    Name = "${var.project}"
  }
}
```

I followed terraform documentation : 

https://www.terraform.io/docs/providers/aws/r/elb.html

Thank's

**Comment by fbentouati** [comment]
When I remove all terraform directory (.terraform, terraform.tfstate), I got a new error : 

```
Error: Error applying plan:

1 error(s) occurred:

* osc_elb.elb-web: 1 error(s) occurred:

* osc_elb.elb-web: Failure adding new or updated ELB listeners: ValidationError: ORN  is not valid.
        status code: 400, request id:

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

My elb is present (health check, listeners is here). 

The backend instance is not associate to this elb.

**Comment by remijouannet** [comment]
fix in v0.4 see the folder examples
